### DISCUSSION TITLE
Move element bounds calculations out of container-block?

### DISCUSSION THREAD

**Discussion by J4v4Scr1pt** [pmndrs/use-gesture#610] [Q&A]
Hi!

This may be a really stupid question but Im calculating the `bounds ` on a component I'm creating. This works as long as I dont place the component inside some div deeper down the page Html hierarchy. Then the `bounds `  is calculated based on it's closest containing block. But my component uses position: fixed, this should take it out of the document flow, and no other parent element has a breaking css (like transform, filter, etc.) 
Is there any way of telling useGesture to base it bounds calculation on the window/viewport rather than its closest parent container?

This is a big component will provide the necessary parts of it.

```
import { createUseGesture, dragAction, hoverAction } from "@use-gesture/react";
import { useTransition, animated, useSpring, config } from "@react-spring/web";
import { useFloating, shift, offset, flip } from "@floating-ui/react-dom";
...other imports

const Wrapper = styled(animated.div)`
	display: flex;
	cursor: pointer;
	z-index: 99999;
	position: fixed;
	touch-action: none;
	user-select: none;
	box-shadow: ${shadow.overlay};
	height: ${spacingIndex * 7}px;
	width: ${spacingIndex * 7}px;
	border-radius: 50%;
	border: 4px solid ${colors.white};
	justify-content: center;
	align-items: center;
`;

....... other code

const DraggableButton = ({ draggableButtonConfig }) => {
        const winX = window.innerWidth || 0;
	const winY = window.innerHeight || 0;
		........... other code
		
	const [{ wrapperX, wrapperY, wrapperScale, wrapperBackground }, wrapperApi] = useSpring(() => ({
		wrapperX: winX && winX > 100 ? winX - 100 : 0,
		wrapperY: winY && winY > 170 ? winY - 170 : 0,
		wrapperScale: 1,
		wrapperBackground: colors.white,
	}));
	........... other code
	
	const bind = useGesture(
       
       ....... other settings

	{
			drag: {
				from: () => [wrapperX.get(), wrapperY.get()],
				bounds: () => {
					const vw = window.innerWidth || 0;
					const vh = window.innerHeight || 0;
					return { left: 0, right: vw - 100, top: 0, bottom: vh - 100 };
				},
				rubberband: true,
				filterTaps: true,
			},
		}
	)
		........... other code
	
	return (
		<>
			{transitionsButton(
				(styles, item) =>
					item && (
						<Wrapper
							{...bind()}
							style={{
								x: wrapperX,
								y: wrapperY,
								scale: open ? 0.9 : wrapperScale,
								background: open
									? draggableButtonConfig.buttonHoverBackgroundColor || colors.lightYellow
									: wrapperBackground,
								...styles,
							}}
							ref={reference}>
						....... other code
```

This image shows the parent containers and where the components calculations is based on:
![image](https://github.com/pmndrs/react-spring/assets/5929055/af885bf2-9094-4f39-8116-c378c70cb2f1)

```
<body>
	<div id="Page">
		<DraggableButton />
		<div id="MainContent">
			<DraggableButton />
			<div id="SubContent">
				<DraggableButton />
			</div>
		</div>
	</div>
</body>
```

If there is anything other I need to provide or I explained badly please let me know 😊!

**Comment by dbismut** [comment]
Hi, you'll have more chances that someone answers if you format your code using syntax highlighting and even more if you provide a sandbox.

**Reply by J4v4Scr1pt** [reply]
Thx, will fix this :)

**Comment by J4v4Scr1pt** [comment]
Working sandbox: 
https://codesandbox.io/s/cranky-yonath-cxz3f2?file=/src/styles.css

The other two draggbleButtons is ouside of the main page and you need to zoom out to see them.
![image](https://github.com/pmndrs/use-gesture/assets/5929055/46665106-bd3a-49fe-9285-115841355404)
![image](https://github.com/pmndrs/use-gesture/assets/5929055/c89bf857-9ed3-4221-bb1d-e07fbf4bffda)
